srml/consensus/src/lib.rs

```diff
@@ -14,7 +14,89 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Consensus module for runtime; manages the authority set ready for the native code.
+//! ## Overview
+//! 
+//! The consensus module manages the authority set for the native code. It provides support for reporting offline behavior among validators and logging changes in the validator authority set.
+//!
+//! ## Interface
+//! 
+//! ### Dispatchable
+//! 
+//! The `Call` enum is documented [here](https://crates.parity.io/srml_consensus/enum.Call.html).
+//! 
+//! - `report_misbehavior` - Report some misbehaviour. The origin of this call must be signed.
+//! - `note_offline` - Note the previous block's validator missed their opportunity to propose a block. The origin of this call must be an inherent.
+//! - `remark` - Make some on-chain remark. The origin of this call must be signed.
+//! - `set_heap_pages` - Set the number of pages in the WebAssembly environment's heap.
+//! - `set_code` - Set the new code.
+//! - `set_storage(Vec<KeyValue>)` - Set some items of storage.
```

Comment:
We should not be providing input types here, will be too easy to go out of sync with changes to the code.